### PR TITLE
Update README.md and UPGRADING.md for 3.0.0 release

### LINKED ISSUE #240: Update README.md with changes to pushgateway client
Putting this here as a reminder for me to do this before we cut a 3.0.0 release.

### PR DESCRIPTION
**Pull request by Sinjo**
We've made a bunch of breaking changes recently, which will be part of
the 3.0.0 release. This commit updates our README to match the new API
in anticipation of that relese.

---

Fixes #240

### CODE REVIEW

**Comment by Sinjo** [comment]
Yeah, let's go for it. I was on the fence since we didn't for the last breaking release (2.0.0), but it seems like a nice thing to do for users.

**Comment by Sinjo** [comment]
bruh

image

**Comment by Sinjo** [comment]
@dmagliola I've pushed what I think is a complete set of notes on breaking changes in the 3.x.x series. Should be ready for a once over!

File: UPGRADING.md [hunk 1]
```diff
@@ -1,3 +1,168 @@
...
+[compatibility policy](COMPATIBILITY.md).
+
+## Push client improvements
+
+### Keyword arguments
+
+In line with changes we made for the 0.10.0 release (see below),
+`Prometheus::Client::Push` now favours the use of keyword arguments for improved clarity
+at the callsites. Specifically, the constructor now takes several keyword arguments rather
+than relying entirely on keyword arguments. Where you would previously have written:
```
Comment (dmagliola) [inline UPGRADING.md:22]: on positional arguments?

**Review by dmagliola — APPROVED** [review]
One tiny typo, otherwise :chefs_kiss:

File: UPGRADING.md (hunk 1 above)
Comment (Sinjo) [reply UPGRADING.md:22]: Good spot!

**Pull request merged**